Src/FluentAssertions/Specialized/ExceptionAssertions.cs

```diff
@@ -191,6 +191,54 @@ public ExceptionAssertions(IEnumerable<TException> exceptions)
             return this;
         }
 
+        private IEnumerable<Exception> AssertInnerExceptionExactly(Type innerException, string because = null,
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(innerException, nameof(innerException));
+
+            AssertInnerExceptions(innerException, because, becauseArgs);
+
+            Exception[] expectedExceptions = Subject
+                .Select(e => e.InnerException)
+                .Where(e => e?.GetType() == innerException).ToArray();
+
+            Execute.Assertion
+                .ForCondition(expectedExceptions.Any())
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected inner {0}{reason}, but found {1}.", innerException, SingleSubject.InnerException);
+
+            return expectedExceptions;
+        }
+
+        private IEnumerable<Exception> AssertInnerExceptions(Type innerException, string because = null,
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(innerException, nameof(innerException));
+
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected inner {0}{reason}, but ", innerException)
+                .ForCondition(Subject is not null)
+                .FailWith("no exception was thrown.")
+                .Then
+                .ForCondition(Subject.Any(e => e.InnerException is not null))
+                .FailWith("the thrown exception has no inner exception.")
+                .Then
+                .ClearExpectation();
+
+            Exception[] expectedInnerExceptions = Subject
+                .Select(e => e.InnerException)
+                .Where(e => e?.GetType() == innerException)
```

Comment:
Hmm, I missed that and thought this code merely moved.

Reply:
It is one of the few changes, so that it would work for both call types. Unfortunately I did not fully grasp the full extent of the OfType<> method. Will fix it!